### PR TITLE
[INLONG-2496][Feature][Sort] Support COS  filesystem when using hive sink

### LINKED ISSUE #2496: [Feature][Sort]Support COS filesystem when using hive sink
### Description

Currently, only hadoop filesystem is supported when using hive sink.
We hope to support more filesystems such as COS and CHDFS

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by chantccc**
### Title Name: [INLONG-XYZ][component] Title of the pull request

where *XYZ* should be replaced by the actual issue number.

Fixes #2496 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by dockerzhang** [comment]
@chantccc thanks for this contribution.  what are `COS` and `CHDFS`, please add more information for them.

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/configuration/Constants.java [hunk 1]
```diff
@@ -336,4 +336,33 @@
     public static final ConfigOption<Integer> ORC_SINK_BATCH_SIZE =
             key(HIVE_SINK_PREFIX + "orc.row.batch.size")
                     .defaultValue(64);
+
+    // ------------------------------------------------------------------------
+    //  Filesystem related
+    // ------------------------------------------------------------------------
+    public enum FilesystemType {
```
Comment (dockerzhang) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/configuration/Constants.java:343]: about `COS` and `CHDFS`, please add more information. what filesystem they are, and where we can get the details.

**Review by KevinWen007 — APPROVED** [review]

**Review by gosonzhang — APPROVED** [review]

**Review by aloyszhang — CHANGES_REQUESTED** [review]
Why commit a Jar file?

**Review by gosonzhang — CHANGES_REQUESTED** [review]
If it is a third-party package required by the project, you need to review its license and add it to the project as a dependency, not as a file

**Pull request closed without merging**